assets/js/components/data-block.js

```diff
@@ -124,7 +125,8 @@ class DataBlock extends Component {
 								/>
 							</span>
 							<span className="googlesitekit-data-block__value">
-								{ `${ numberFormat( Math.abs( change ) ) }${ changeDataUnit } ${ period }` }
+								{ period && sprintf( period, `${ numberFormat( Math.abs( change ) ) }${ changeDataUnit }` ) }
+								{ ! period && `${ numberFormat( Math.abs( change ) ) }${ changeDataUnit }` }
```

Comment:
I've modified this by trying to format as a currency, which will solve the currency formatting issue (good catch!). I've also removed usage of hard-coded `$`s, that's a nice improvement we hadn't spotted before!

As far as the other potential units go, I think we shouldn't have a generic localized concatenation. Since we only have `%` for now, I don't think it's relevant - maybe we can revisit that part later.